rust/src/detect/transform_base64.rs

```diff
@@ -0,0 +1,420 @@
+                let s_val = unsafe { CStr::from_ptr(self.nbytes_str) };
+                let o_val = unsafe { CStr::from_ptr(other.nbytes_str) };
+                res = s_val == o_val;
+            } else if !self.nbytes_str.is_null() || !other.nbytes_str.is_null() {
+                return false;
+            }
+
+            if !self.offset_str.is_null() && !other.offset_str.is_null() {
+                let s_val = unsafe { CStr::from_ptr(self.offset_str) };
+                let o_val = unsafe { CStr::from_ptr(other.offset_str) };
+                res = s_val == o_val;
+            } else if !self.offset_str.is_null() || !other.offset_str.is_null() {
+                return false;
+            }
+
+            res && self.nbytes == other.nbytes
+                && self.flags == other.flags
+                && self.offset == other.offset
+                && self.mode == other.mode
+        }
+    }
+
+    fn valid_test(
+        args: &str,
+        nbytes: u32,
+        nbytes_str: &str,
+        offset: u32,
+        offset_str: &str,
+        mode: DetectBase64Mode,
+        flags: u8,
+    ) {
+        let tbd = DetectTransformFromBase64Data {
+            flags,
+            nbytes,
+            nbytes_str: if !nbytes_str.is_empty() {
+                CString::new(nbytes_str).unwrap().into_raw()
+            } else {
+                std::ptr::null_mut()
+            },
+            offset,
+            offset_str: if !offset_str.is_empty() {
+                CString::new(offset_str).unwrap().into_raw()
+            } else {
+                std::ptr::null_mut()
+            },
+            mode,
+        };
+
+        let (_, val) = parse_transform_base64(args).unwrap();
+        assert_eq!(val, tbd);
+    }
+
+    #[test]
+    fn test_parser_invalid() {
+        assert!(parse_transform_base64("bytes 4, offset 3933, mode unknown").is_err());
+        assert!(parse_transform_base64("bytes 4, offset 70000, mode strict").is_err());
+        assert!(
+            parse_transform_base64("bytes 4, offset 70000, mode strict, mode rfc2045").is_err()
+        );
+    }
+
+    #[test]
+    fn test_parser_parse_partial_valid() {
+        let mut tbd = DetectTransformFromBase64Data {
+            nbytes: 4,
+            offset: 0,
+            mode: TRANSFORM_FROM_BASE64_MODE_DEFAULT,
+            flags: 0,
+            ..Default::default()
+        };
+
+        tbd.mode = TRANSFORM_FROM_BASE64_MODE_DEFAULT;
+        tbd.flags = DETECT_TRANSFORM_BASE64_FLAG_NBYTES;
+        let (_, val) = parse_transform_base64("bytes 4").unwrap();
+        assert_eq!(val, tbd);
+
+        tbd.offset = 3933;
+        tbd.flags = DETECT_TRANSFORM_BASE64_FLAG_NBYTES | DETECT_TRANSFORM_BASE64_FLAG_OFFSET;
+        let (_, val) = parse_transform_base64("bytes 4, offset 3933").unwrap();
+        assert_eq!(val, tbd);
+
+        tbd.flags = DETECT_TRANSFORM_BASE64_FLAG_NBYTES | DETECT_TRANSFORM_BASE64_FLAG_OFFSET;
```

Comment:
Unneeded line ?

Reply:
I'm going to leave this for clarity purposes.